cpu/stm32f1/periph/adc.c

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2015 Engineering-Spirit
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     cpu_stm32f1
+ * @{
+ *
+ * @file
+ * @brief       Low-level ADC driver implementation
+ *
+ * @author      Nick van IJzendoorn <nijzendoorn@engineering-spirit.nl>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+#include "periph/adc.h"
+#include "periph_conf.h"
+
+/* guard in case that no ADC device is defined */
+#if ADC_NUMOF
```

Comment:
I test it on a board we want to put on KickStarter before x-mas.

The defines I use in `periph_conf.h` are:

	/**
	 * @name ADC configuration
	 * @{
	 */
	#define ADC_NUMOF           (1U)
	#define ADC_0_EN            1
	#define ADC_MAX_CHANNELS    17

	/* ADC 0 configuration */
	#define ADC_0_DEV           ADC1
	#define ADC_0_CHANNELS      17
	#define ADC_0_CLKEN()       (RCC->APB2ENR |= RCC_APB2ENR_ADC1EN)
	#define ADC_0_CLKDIS()      (RCC->APB2ENR &= ~(RCC_APB2ENR_ADC1EN))
	#define ADC_0_PORT_CLKEN()  (RCC->APB2ENR |= RCC_APB2ENR_IOPAEN | RCC_APB2ENR_IOPBEN | RCC_APB2ENR_IOPCEN)
	/** @} */

The pin defines can be found in in the data sheets of the STM32F1xx MCU's under section 3 (Pinouts and pin description) Table 5. (....-density STM32F103xx pin definitions (continued)) From there out you can search for ADC12_IN##. The idea is that ADC channels 0 -> 7 are mapped on PA0 -> PA7 channels 8 & 9 are mapped on PB0 & PB1 and channels 10 -> 15 on PC0 -> PC5, Channel 16 and 17 are the internal channels for the temperature and the reference. I belief this to be equal for ALL STM's since STM wants to keep there processors drop in replaceable.

Multiple ADC device is possible, but since the read function is blocking the readouts can't interleave I think?